### PR TITLE
.deb, .rpm and general linux support + a little crossplatform

### PR DESCRIPTION
**Pull request by Voodlaz**
This PR will add package data for cargo-deb util to create an installable deb package, and general linux and crossplatfrom support(crossplatfrom have some not done yet parts). This includes sections in deb packages, crossplatfrom config and data directories putting assets in the right place. This partly solve this issue: #24

### CODE REVIEW

**Comment by Voodlaz** [comment]
we'll still need a normal icon..

**Comment by Voodlaz** [comment]
I made the icon, now it's good to go.

**Comment by Voodlaz** [comment]
oh dear, I forgot about dependencies.

**Comment by Voodlaz** [comment]
I added rpm

**Comment by Voodlaz** [comment]
@mrDIMAS , okay, I think I;ve done everything.

**Comment by mrDIMAS** [comment]
In general it looks fine, but I please run `clippy` to make code more idiomatic - for example you have `return` statements at the end of functions, checking boolean statements with `== true` or `== false` which is not wrong, but a bit useless, and some more tiny issues. 

Also, why do you need to change path to resources? Maybe it is worth to just use `include_bytes!` macro to embed resources in the final binary?

**Comment by Voodlaz** [comment]
@mrDIMAS 

- ran clippy suggestions
- about that. existence resources is an easy way to check whetter the programm is installed or not.

**Comment by Voodlaz** [comment]
> Also, why do you need to change path to resources? Maybe it is worth to just use `include_bytes!` macro to embed resources in the final binary?

Wouldn't it be more neat to use resources inside resources folder that standarts make?

**Review by mrDIMAS — CHANGES_REQUESTED** [review]
Please fix the commented lines.

File: src/custom_dirs.rs [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+        path.push(filename);
+
+        path
+    } else {
+        let startup_working_dir = debug_dir(filename);
+        startup_working_dir
+    }
+}
+
+pub fn resources_dir(category: &str, filename: &str) -> PathBuf {
```
[the remark this answers is not in the document]
Comment (Voodlaz) [reply src/custom_dirs.rs:46]: great! I see you embeded that font into the binary!

[the remark this answers is not in the document]
Comment (Voodlaz) [reply src/custom_dirs.rs:46]: will remove it.

File: src/custom_dirs.rs [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+        let project_dir = ProjectDirs::from("org","rg3dengine", "rusty-editor");
+        let dirs = project_dir.unwrap().data_dir().to_owned();
+
+        let mut path = PathBuf::new();
+        path.push(dirs);
+        path.push(filename);
+
+        path
+    } else {
+        let startup_working_dir = debug_dir(filename);
```
[the remark this answers is not in the document]
Comment (Voodlaz) [reply src/custom_dirs.rs:41]: okay

File: src/custom_dirs.rs [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+        let project_dir = ProjectDirs::from("org","rg3dengine", "rusty-editor");
+        let dirs = project_dir.unwrap().config_dir().to_owned();
+
+        let mut path = PathBuf::new();
+        path.push(dirs);
+        path.push(filename);
+
+        path
+    } else {
+        let startup_working_dir = debug_dir(filename);
```
[the remark this answers is not in the document]
Comment (Voodlaz) [reply src/custom_dirs.rs:25]: okay

File: src/custom_dirs.rs [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+/*
+The purpose of this file is using directories-rs library
+with a little bit of customization. This includes checking for existance
+and adding file to the end of the path.
+*/
+
+pub fn config_dir(filename: &str) -> PathBuf {
+    if *TEST_EXISTENCE {
+        let project_dir = ProjectDirs::from("org","rg3dengine", "rusty-editor");
```
[the remark this answers is not in the document]
Comment (Voodlaz) [reply src/custom_dirs.rs:16]: okay

File: src/custom_dirs.rs [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+        path
+    } else {
+        let startup_working_dir = debug_dir(filename);
+        startup_working_dir
+    }
+}
+
+pub fn other_dir(filename: &str) -> PathBuf {
+    if *TEST_EXISTENCE {
+        let project_dir = ProjectDirs::from("org","rg3dengine", "rusty-editor");
```
[the remark this answers is not in the document]
Comment (Voodlaz) [reply src/custom_dirs.rs:32]: okay

File: src/main.rs [hunk 6]
```diff
@@ -137,10 +141,46 @@ pub fn send_sync_message(ui: &Ui, mut msg: UiMessage) {
...
+    };
+    // this constant tests if programm directories exist,
+    // which means that programm is installed, and not running
+    static ref TEST_EXISTENCE: bool = {
+        // check for debug handle
+        if !(*DEBUG_HANDLE) {
+            let path = custom_dirs::other_dir_test("");
+            let test = Path::new(&path).exists();
+
+            if test {
```
[the remark this answers is not in the document]
Comment (Voodlaz) [reply src/main.rs:168]: okay

File: src/custom_dirs.rs [hunk 7]
```diff
@@ -0,0 +1,107 @@
...
+
+    let mut path = PathBuf::new();
+    path.push(dirs);
+    path.push(filename);
+
+    path
+}
+
+pub fn other_dir_test(filename: &str) -> PathBuf {
+    let project_dir = ProjectDirs::from("org","rg3dengine", "rusty-editor");
```
[the remark this answers is not in the document]
Comment (Voodlaz) [reply src/custom_dirs.rs:89]: okay

File: src/custom_dirs.rs [hunk 8]
```diff
@@ -0,0 +1,107 @@
...
+        let mut startup_working_dir = debug_dir("resources");
+        startup_working_dir.push(category);
+        startup_working_dir.push(filename);
+
+        startup_working_dir
+    }
+}
+
+fn debug_dir(filename: &str) -> PathBuf {
+    let dirs = std::env::current_dir().unwrap();
```
[the remark this answers is not in the document]
Comment (Voodlaz) [reply src/custom_dirs.rs:65]: okay

File: src/custom_dirs.rs [hunk 9]
```diff
@@ -0,0 +1,107 @@
...
+    path.push(dirs);
+    path.push(filename);
+
+    path
+}
+
+// these are only used for creating this directories, and a checking inside
+// TEST_EXISTENCE constant. Because normal ones is using this constant value,
+// but the constant haven't returned the value yet, so it crashes.
+pub fn config_dir_test(filename: &str) -> PathBuf {
```
[the remark this answers is not in the document]
Comment (Voodlaz) [reply src/custom_dirs.rs:77]: okay

**Comment by Voodlaz** [comment]
whoops, accidently pressed reviewer needed button

File: src/custom_dirs.rs [hunk 10]
```diff
@@ -0,0 +1,107 @@
...
+    let dirs = project_dir.unwrap().data_dir().to_owned();
+
+    let mut path = PathBuf::new();
+    path.push(dirs);
+    path.push(filename);
+
+    path
+}
+
+pub fn resources_dir_test(filename: &str) -> PathBuf {
```
[the remark this answers is not in the document]
Comment (mrDIMAS) [reply src/custom_dirs.rs:59]: Remove this function, it is not used anyway.

**Review by mrDIMAS — CHANGES_REQUESTED** [review]
Please run `cargo fmt` to unify code formatting. Also use a spell checker, there's lots of grammatical errors in comments.

File: os-installation/general-description.txt [hunk 11]
```diff
@@ -0,0 +1,5 @@
+A scene editor for rg3d game engine.
+
+rg3d is a feature rich game egine made in rust, which allows
```
Comment (mrDIMAS) [inline os-installation/general-description.txt:3]: typo `egine`

File: os-installation/general-description.txt [hunk 12]
```diff
@@ -0,0 +1,5 @@
...
+you to build shooters, platfomers, sandoxes and more
```
Comment (mrDIMAS) [inline os-installation/general-description.txt:4]: typo `sandoxes`

File: resources/README.md [hunk 13]
```diff
@@ -0,0 +1,3 @@
+- embed - for stuff like icons and fonts that are embeded into binary
```
Comment (mrDIMAS) [inline resources/README.md:1]: typo `embeded`

File: resources/README.md [hunk 14]
```diff
@@ -0,0 +1,3 @@
...
+- lib - for other stuff that couldn't be embeded into binary, but have to be used rust code like embadble icons.
```
Comment (mrDIMAS) [inline resources/README.md:2]: typos `embeded` and `embadble`

File: resources/README.md [hunk 15]
```diff
@@ -0,0 +1,3 @@
...
+- etc - for everything that dosen't fit into this categories. for example, icons for system launchpad.
```
Comment (mrDIMAS) [inline resources/README.md:3]: typo `dosen't`

File: resources/lib/release.toml [hunk 16]
```diff
@@ -0,0 +1,14 @@
+# Use this switch only for development or debugging purposes.
+# Do not switch the switch inside cargo project, it's usefull
```
Comment (mrDIMAS) [inline resources/lib/release.toml:2]: typo `usefull`

File: resources/lib/release.toml [hunk 17]
```diff
@@ -0,0 +1,14 @@
...
+# switching only in installed programm.
+#
+# Switching the switch will make programm generate new files
+# inside cargo project, instead directories that gives
+# OS to installed programms, like ProgamFiles on windows
```
Comment (mrDIMAS) [inline resources/lib/release.toml:7]: typo `ProgamFiles`

**Review by mrDIMAS — CHANGES_REQUESTED** [review]
Please fix typos and then I'll merge the PR.

**Pull request merged**